spec/couchbase/protostellar/management/collection_manager_spec.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+#  Copyright 2023. Couchbase, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+require "rspec"
+
+RSpec.describe Couchbase::Protostellar::Management::CollectionManager do
+  before(:all) do
+    cluster = connect
+    @collection_mgr = test_bucket(cluster).collections
+  end
+
+  def get_collection_spec(scope_name:, collection_name:)
+    Couchbase::Management::CollectionSpec.new do |spec|
+      spec.name = collection_name
+      spec.scope_name = scope_name
+    end
+  end
+
+  describe "#get_all_scopes" do
+    subject(:scopes) { @collection_mgr.get_all_scopes }
+
+    it "returns an array" do
+      expect(scopes.class).to be Array
+    end
+  end
+
+  describe "#create_scope" do
+    context "when the scope does not currently exist" do
+      it "creates the scope" do
+        scope_name = unique_id(:scope).tr('.', '')
+        @collection_mgr.create_scope(scope_name)
+        expect(@collection_mgr.get_all_scopes.any? { |scope_spec| scope_spec.name == scope_name }).to be true
+      end
+    end
+
+    context "when the scope already exists" do
+      it "raises a ScopeExists error" do
+        scope_name = unique_id(:scope).tr('.', '')
```

Comment:
could you extract this `unique_id(:scope).tr('.', '')` into some helper? or just fix the `unique_id` to generate name as safe as possible.